components/telemetry-operator/internal/parsers/parser_test.go

```diff
@@ -0,0 +1,36 @@
+package parsers
+
+import (
+	"context"
+	"testing"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+	"github.com/kyma-project/kyma/components/telemetry-operator/internal/sync/mocks"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var daemonSetConfig = FluentBitDaemonSetConfig{
+	FluentBitDaemonSetName:    types.NamespacedName{Name: "telemetry-fluent-bit", Namespace: "cm-ns"},
+	FluentBitParsersConfigMap: types.NamespacedName{Name: "telemetry-fluent-bit", Namespace: "cm-ns"},
+}
+
+func TestSyncParsersConfigMapErrorClientErrorReturnsError(t *testing.T) {
```

Comment:
We should also test the successful case. Let's use the [controller-runtime fake client](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.3/pkg/client/fake).